java/clients/src/test/java/sleeper/clients/QueryWebSocketClientTest.java

```diff
@@ -282,12 +324,14 @@ void shouldHandleErrorIfMessageWithErrorIsReceived() throws Exception {
             // Given
             Query query = exactQuery("test-query-id", 123L);
 
-            // When
-            runQuery(query,
+            // When / Then
+            assertThat(runQueryFuture(query,
                     withResponses(
-                            message(errorMessage("test-query-id", "Query failed"))));
-
-            // Then
+                            message(errorMessage("test-query-id", "Query failed")))))
```

Comment:
Looking into this test case, I noticed that this doesn't involve a call to close the connection.

It looks like right now WebSocketMessageHandler.onClose is only called if it's specifically added to the list of responses. It should probably also be called in closeBlocking if it's not already closed? It looks like that's what will happen in the real web socket implementation?

Reply:
Looking again, it doesn't seem quite valid that the WebSocketResponse objects operate directly on the WebSocketMessageHandler. We're asserting on the connected and closed state in the FakeWebSocketClient, but the WebSocketResponse operations bypass that and go straight to the WebSocketMessageHandler. Should they go to the FakeWebSocketClient instead?